lib/modules.nix

```
@@ -116,6 +129,75 @@ rec {
               turned off.
             '';
           };
+
+          _module.checks = mkOption {
+            description = ''
+              Evaluation checks to trigger during module evaluation. The
+              attribute name will be displayed when it is triggered, allowing
+              users to disable/change these checks if necessary. See
+              the section on Warnings and Assertions in the manual for more
+              information.
+            '';
+            example = literalExample ''
+              {
+                gpgSshAgent = {
+                  enable = config.programs.gnupg.agent.enableSSHSupport && config.programs.ssh.startAgent;
+                  message = "You can't use ssh-agent and GnuPG agent with SSH support enabled at the same time!";
+                };
+
+                grafanaPassword = {
+                  enable = config.services.grafana.database.password != "";
+                  message = "Grafana passwords will be stored as plaintext in the Nix store!";
+                  type = "warning";
+                };
+              }
+            '';
+            default = {};
+            internal = prefix != [];
+            type = types.attrsOf (types.submodule {
+              options.enable = mkOption {
+                description = ''
+                  Whether to enable this check.
+                '';
+                default = true;
+                type = types.bool;
+              };
+
+              options.assertion = mkOption {
```

```suggestion
              options.check = mkOption {
```

The term `assertion` is incompatible with `type = "warning"`.
Normally an assertion means that a condition _must_ hold, with the effect of program cancellation when it turns out not to.

---

Yeah I guess that's a good point, I'll change it to `check`